dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp

```diff
@@ -170,10 +171,16 @@ Block CHBlockChunkCodec::decodeImpl(ReadBuffer & istr, size_t reserve_size)
 
         /// Data
         MutableColumnPtr read_column = column.type->createColumn();
-        if (reserve_size > 0)
-            read_column->reserve(std::max(rows, reserve_size));
-        else if (rows)
-            read_column->reserve(rows);
+        /// For non-fixed size type, reserve function might cause too much memory usage, i.e. string type column reserves 64 bytes
+        /// size for each element.
+        const auto & type_removed_nullable = removeNullable(column.type);
+        if (type_removed_nullable->isValueRepresentedByNumber() || type_removed_nullable->isFixedString())
```

Comment:
Maybe add a function `reservable`, and add some comments why some data types cannot be reserved.

Reply:
Make sense, I'll add it

Reply:
Found one existing interface "haveMaximumSizeOfValue" exactly match the request. Use it instead.